service-config/src/main/java/com/palantir/config/service/ServiceConfiguration.java

```diff
@@ -44,22 +44,10 @@
      */
     public abstract List<String> uris();
 
-    // hides implementation details
     public static Builder builder() {
-        return ImmutableServiceConfiguration.builder();
+        return new Builder();
     }
 
-    // hides implementation details
-    public interface Builder {
-
-        Builder apiToken(BearerToken apiToken);
-
-        Builder security(SslConfiguration security);
-
-        Builder uris(Iterable<String> uris);
-
-        Builder from(ServiceConfiguration otherConfig);
-
-        ServiceConfiguration build();
+    public static final class Builder extends ImmutableServiceConfiguration.Builder {
```

Comment:
* I dislike referring directly to generated types because you can no longer find them in github, etc. Generated methods are fine as one once you've found the type, you can see the code generation annotations and look up the docs.
* Exposing final types in your API makes downstream tests fragile, as Mockito can no longer create mocks. [Smart nulls](http://docs.mockito.googlecode.com/hg/1.9.5/org/mockito/Answers.html#RETURNS_SMART_NULLS) are super helpful in tracking down test failures, and being able to pass trivial mocks through wiring code means simpler, clearer tests.

FWIW, you can keep your static `.of` methods, as static methods are inherited in Java, though IDEs aren't always cooperative.